test/fixtures/rules/std_rule_cases/L042.yml

```diff
@@ -7,17 +7,143 @@ select_fail:
     from a
     join (
         select x, z from b
-    ) on (a.x = b.x)
+    ) as b on (a.x = b.x)
+  fix_str: |
+    with b as (
+        select x, z from b
+    )
+    select
+        a.x, a.y, b.z
+    from a
+    join b on (a.x = b.x)
+
+cte_select_fail:
+  fail_str: |
+    with prep as (
+      select 1 as x, 2 as z
+    )
+    select
+        a.x, a.y, b.z
+    from a
+    join (
+        select x, z from b
+    ) as b on (a.x = b.x)
+  fix_str: |
+    with prep as (
+      select 1 as x, 2 as z
+    ),
+    b as (
+        select x, z from b
+    )
+    select
+        a.x, a.y, b.z
+    from a
+    join b on (a.x = b.x)
+
+double_nested_fail:
+  fail_str: |
+    with p_cte as (
+      select 1 as x, 2 as z
+      UNION ALL
+      select 1 as x, 2 as z
+    )
+    select
+        a.x, a.y, b.z
+    from a
+    join (
+      select x, z from (
+        select x, z from p_cte
+      ) as c
+    ) as b on (a.x = b.x)
+  fix_str: |
+    with p_cte as (
+      select 1 as x, 2 as z
+      UNION ALL
+      select 1 as x, 2 as z
+    ),
+    c as (
+        select x, z from p_cte
+      ),
```

Comment:
Indentation here comes back as it was (the indentation of the origin was correct for a subquery, but is now incorrect) will another pass/Rule pick this up?

Reply:
Yes, if you run `sqlfluff lint`, the rules work "together" to provide reasonable output. We've discussed this for other rules, and for now, we generally agree it's better to let other rules handle it rather than have each rule doing "all the things". (We may change this someday, but there's no clear timeline for doing that. The current way works pretty well in practice.)